.circleci/config.yml

```diff
@@ -5,194 +5,98 @@
+                  echo "Merging $(cat merge.txt)";
+                  git pull --ff-only upstream "refs/pull/$(cat merge.txt)/merge";
+              fi
       - run:
-          name: Create new conda env
+          name: setup conda
           command: |
-            if
-              conda env list | grep testenv
-            then
-              echo "Conda env testenv already exists courtesy of the cache."
-            else
-              conda create -n testenv -yq
-            fi
-
-  cache_ignorant_conda_setup:
-    description: "Downloads & installs only the fresh copy of conda."
-    steps:
+              wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh;
+              chmod +x ~/miniconda.sh && ~/miniconda.sh -b;
+              echo 'export PATH="$HOME/miniconda3/bin:$PATH"'  >> $BASH_ENV;
       - run:
-          name: setup conda afresh
+          name: Install apt packages
           command: |
-            wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
-            chmod +x ~/miniconda.sh && ~/miniconda.sh -b
-            echo 'export PATH="$HOME/miniconda3/bin:$PATH"'  >> $BASH_ENV
-      - run: conda create -n testenv
-
-  install_dependencies:
-    description: "Installs the necessary Python packages"
-    steps:
+              ./tools/circleci_dependencies_apt.sh
       - run:
           name: Install packages in conda env
           command: |
-            conda install -n testenv python=3.7 numpy scipy scikit-learn matplotlib pandas \
-            lxml mkl sphinx numpydoc pillow pandas -yq
-            conda install -n testenv nibabel sphinx-gallery junit-xml -c conda-forge -yq
-
-  build_docs:
-    description: "Installs Nilearn & builds documentation using Sphinx's make html-strict"
-    steps:
+              ./tools/circleci_dependencies.sh
       - run:
-          name: Building documentation
+          name: Find build type
           command: |
-            source activate testenv
-            pip install -e .
-            set -o pipefail && cd doc && make html-strict 2>&1 | tee log.txt
+              ./tools/circleci_build_type.sh
+      - run:
+          name: Verify build type
+          command: |
+              echo "PATTERN = $(cat pattern.txt)"
+              echo "BUILD = $(cat build.txt)"
+      - run:
+          name: build doc
+          command: |
+              source activate testenv;
+              echo "Conda active env = $CONDA_DEFAULT_ENV";
+              cd doc;
+              PATTERN=$(cat ../pattern.txt) make $(cat ../build.txt)
           no_output_timeout: 7h
-
-  store_results:
-    description: "Stores build times and artifacts"
-    steps:
       - store_test_results:
           path: doc/_build/test-results
       - store_artifacts:
           path: doc/_build/test-results
+          destination: test-results
+      - store_artifacts:
+          path: doc/_build/html/
```

Comment:
in [the build](https://app.circleci.com/pipelines/github/nilearn/nilearn/1718/workflows/03041277-3bcb-486c-be19-95865f02c932/jobs/6892) it says it doesn't find the artifacts?

Reply:
I had a closer look at these upload steps:

- I think the `test_results` storing step doesn't have anything to store since it is not a full build (I'll request a full build to double check that)
- uploading the dev documentation works properly
- uploading the stable documentation doesn't do anything since we're not hitting a stable target of the makefile
- Unless I'm missing something, it looks like the coverage step is not doing anything on the current circleCI workflow either. I will try to see if I can make it work.
- the file `log.txt` was indeed deleted at some point. Thanks for catching this! I fixed that in commit ba164c8.

Reply:
great, thanks! I think I would remove the coverage step; I don't think it's relevant for the documentation build